### ISSUE TITLE
[BQ Aquaris X2] [Android P GSI] Boots, but various broken peripherals

### ISSUE THREAD

**Issue by sylentprofet** [phhusson/treble_experimentations#358]
BQ Aquaris X2 boots to home screen, and some things work (touchscreen, wifi) but various peripherals are broken in Android 9.

It used to work perfectly on previous builds (8.1). Using a CAF Android 9 userdebug boot and vendor.
Tried both the prepackaged arm64-ab-gapps-su and compiling it myself from source, same issues for both.

- No camera
- No sound
- No cell radio / SIM not detected (but wifi works!)

Attached is a logcat, with a brief snippet below showing the camera specific error.

Any pointers would be greatly appreciated. Thanks for all the hard work @phhusson !

```
04-20 02:08:57.870  1302  1302 I android.hardware.camera.provider@2.4-service: Camera provider Service is starting.
04-20 02:08:57.871  1302  1302 D vndksupport: Loading /vendor/lib/hw/android.hardware.camera.provider@2.4-impl.so from current namespace instead of sphal namespace.
04-20 02:08:57.882  1302  1302 D vndksupport: Loading /vendor/lib/hw/camera.sdm660.so from current namespace instead of sphal namespace.
04-20 02:08:57.883  1302  1302 E HAL     : load: module=/vendor/lib/hw/camera.sdm660.so
04-20 02:08:57.883  1302  1302 E HAL     : dlopen failed: library "android.hidl.base@1.0.so" not found
04-20 02:08:57.883  1302  1302 E CamProvider@2.4-impl: Could not load camera HAL module: -22 (Invalid argument)
04-20 02:08:57.883  1302  1302 E CamProvider@2.4-impl: HIDL_FETCH_ICameraProvider: camera provider init failed!
04-20 02:08:57.885  1302  1302 E android.hardware.camera.provider@2.4-service: Could not get passthrough implementation for android.hardware.camera.provider@2.4::ICameraProvider/legacy/0.
```

Attached logcat:

[gsi.log](https://github.com/phhusson/treble_experimentations/files/2751359/gsi.log)

**Comment by dimusa6** [comment]
@sylentprofet I have a BQ Aquaris X2 (3GB of ram and 32 GB model) and it's work perfectly. I only have to use another camera because stock one doesn't work, but I don't have any problem.
Did you have lastest vendor form BQ? Maybe you can try to flash lastest stock rom, and after try again with phh treble rom

**Comment by Klozz** [comment]
missing : android.hidl.base@1.0.so
missing: vendor.nxp.hardware.nfc@1.0.so
missing: vendor.qti.hardware.radio.qtiradio@2.1.so

copy those libs from system/lib and lib64 to vendor/lib and lib64 (from your treble base rom) then re-flash phh's 
and you will get radio and nfc

linker  : CANNOT LINK EXECUTABLE "/vendor/bin/hw/android.hardware.media.omx@1.0-service": cannot locate symbol "_ZN7android15PermissionCache15checkPermissionERKNS_8String16Eij" referenced by "/vendor/lib/libgui_vendor.so"...

your omx blob have missing symbols (camera not works)

**Comment by sylentprofet** [comment]
@dimusa6 I have tried the following versions:
> From BQ's Downloads, flashed the latest available 1.7.2 (based on Android 8.1, Feb 1 patch level)
> Tried phh 9.0 v110 and 8.1 v28, both result in boot loops. I suspect this might be due to a system patch level mismatch, but I'm not sure.

Can you please tell me which BQ firmware and phh system version specifically worked for you?

@Klozz I will give this a try and report back, thanks!

**Comment by phhusson** [comment]
This issue is specifically because of a community vendor, not stock vendor, so i'm closing this
